addons/account_edi_ubl/models/account_move.py

```diff
@@ -11,6 +11,7 @@ def _get_ubl_values(self):
         def format_monetary(amount):
             # Format the monetary values to avoid trailing decimals (e.g. 90.85000000000001).
             return float_repr(amount, self.currency_id.decimal_places)
+        self.ensure_one()
```

Comment:
That doesn't seem useful.

Reply:
It is, this method returns the ubl values for only one invoice. If it's called on a recordset with multiple records, if might crash (for example, if they have a different move_type).

Reply:
It will crash either way. This makes it crash, not having this will make it crash when `move_type` is accessed. The difference is whether it's called on *no* invoice (an empty recordset).

Reply:
Still, IMO it's clearer to crash on self.ensure_one, and before even crashing just reading the first line of the method makes you notice that this should be called with one record.

Calling it with 0 invoice makes no sense.

Reply:
Fair enough.